src/unix/aix-common.c

```diff
@@ -253,15 +263,16 @@ int uv_interface_addresses(uv_interface_address_t** addresses, int* count) {
     sa_addr = (struct sockaddr_dl*) &p->ifr_addr;
     memcpy(address->phys_addr, LLADDR(sa_addr), sizeof(address->phys_addr));
 
-    if (ioctl(sockfd, SIOCGIFNETMASK, p) == -1) {
-      uv__close(sockfd);
-      return UV_ENOSYS;
-    }
-
-    if (inet6)
-      address->netmask.netmask6 = *((struct sockaddr_in6*) &p->ifr_addr);
-    else
+    if (inet6) {
+      memset((char*) &if6, 0, sizeof(if6));
+      strncpy(if6.ifr_name, p->ifr_name, sizeof(if6.ifr_name) - 1);
+      memcpy(&if6.ifr_Addr, &p->ifr_addr, sizeof(if6.ifr_Addr));
+      if (ioctl(sock6fd, SIOCGIFNETMASK6, &if6) == 0) {
```

Comment:
Since `sock6fd` isn't used until this point, would it be simpler to wait until here to initialize it?